packages/workbox-background-sync/lib/Queue.mjs

```diff
@@ -0,0 +1,297 @@
+/*
+ Copyright 2017 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import {_private} from 'workbox-core';
+import StorableRequest from './StorableRequest.mjs';
+
+
+const {indexedDBHelper, WorkboxError} = _private;
+const names = new Set();
```

Comment:
This feels like an unusual pattern - I think you are doing it to ensure a common set of names across all instances of Queue. Would be good to add a comment

Reply:
What do you mean? Using a set rather than an array?